use same master eval logic for Minion and SMinion

jeremyrosenbaum opened the pull request:
This fixes #24354

There was inconsistent behavior between salt-minion (`Minion` under the hood)
and salt-call (`SMinion` under the hood) with regard to how the master was
determined from the minion config.  `SMinion` had its own hard-coded master
evaluation in its `__init__` method that was not congruent with documentation.
The "standard" master evaluation logic existed in `master_eval`, a method of
`Minion`.
This was resolved by moving `master_eval` into `MinionBase` (which `Minion`
inherits from) and making `SMinion` also subclass `MinionBase`, which
otherwise only consists of a couple of methods that `SMinion` won't access.  This diff looks bigger than it
actually is because I had to move the class definition for `MinionBase` above `SMinion` so it could inherit.

I tested these changes successfully with the 2015.8 branch, verifying that salt-master, salt-minion and salt-call all behave as expected on startup and master connection.

Bonus: A couple of docstring typo fixes I came across while working on this ;)

Commit:
fix some docstring typos

Commit:
use same master eval logic for Minion and SMinion

This fixes https://github.com/saltstack/salt/issues/24354
There was inconsistent behavior between salt-minion (`Minion` under the hood)
and salt-call (`SMinion` under the hood) with regard to how the master was
determined from the minion config.  `SMinion` had its own hard-coded master
evaluation in its `__init__` method that was not congruent with documentation.
The "standard" master evaluation logic existed in `master_eval`, a method of
`Minion`.
This was resolved by moving `master_eval` into `MinionBase` (which `Minion`
inherits from) and making `SMinion` also subclass `MinionBase`, which
otherwise only consists of a couple of methods that `SMinion` won't access.

cachedout commented:
I think this is probably quite safe but it will take me a little while to thoroughly review it. Thanks for sending this over though! This looks like it will be a very good change.

cachedout commented:
@jodv There are two small lint errors. Could you please clean those up while we work on the review? Thanks!

Commit:
Fix lint errors

Running the `MinionBase` init method with `super` is literally the same as the
`self.opts = opts` statement that was in the init method for `SMinion` anyway
so if it makes flake8 happy, then sure.

jodv commented:
Done!  The one complaining about not calling init via `super` was kind of funny because that was on purpose, but upon further review, it's equivalent to just do that anyway (until someone updates `MinionBase`'s init method :wink:)

cachedout commented:
@jodv Thanks! We'll look at this as quickly as possible. Many thanks for your hard work here.

jodv commented:
Thanks for the quick review @cachedout !
